python/cuml/ensemble/randomforestregressor.pyx

```diff
@@ -0,0 +1,620 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# cython: profile=False
+# distutils: language = c++
+# cython: embedsignature = True
+# cython: language_level = 3
+
+import ctypes
+import math
+import numpy as np
+import warnings
+import pdb
+
+from numba import cuda
+
+from libcpp cimport bool
+from libc.stdint cimport uintptr_t
+from libc.stdlib cimport calloc, malloc, free
+
+from cuml.common.base import Base
+from cuml.common.handle cimport cumlHandle
+from cuml.utils import get_cudf_column_ptr, get_dev_array_ptr, \
+    input_to_dev_array, zeros
+cimport cuml.common.handle
+cimport cuml.common.cuda
+
+cdef extern from "randomforest/randomforest.hpp" namespace "ML":
+    cdef enum CRITERION:
+        GINI,
+        ENTROPY,
+        MSE,
+        MAE,
+        CRITERION_END
+
+cdef extern from "decisiontree/decisiontree.hpp" namespace "ML::DecisionTree":
+    cdef struct DecisionTreeParams:
+        int max_depth
+        int max_leaves
+        float max_features
+        int n_bins
+        int split_algo
+        int min_rows_per_node
+        bool bootstrap_features
+        bool quantile_per_tree
+        CRITERION split_criterion
+
+cdef extern from "randomforest/randomforest.hpp" namespace "ML":
+
+    cdef enum RF_type:
+        CLASSIFICATION,
+        REGRESSION
+
+    cdef struct RF_metrics:
+        RF_type rf_type
+        float accuracy
+        double mean_abs_error
+        double mean_squared_error
+        double median_abs_error
+
+    cdef struct RF_params:
+        int n_trees
+        bool bootstrap
+        float rows_sample
+        pass
+
+    cdef cppclass RandomForestMetaData[T, L]:
+        void* trees
+        RF_params rf_params
+
+    # Random Forest Classifier
+    cdef void fit(cumlHandle & handle,
+                  RandomForestMetaData[float, float]*,
+                  float *,
+                  int,
+                  int,
+                  float *,
+                  RF_params) except +
+
+    cdef void fit(cumlHandle & handle,
+                  RandomForestMetaData[double, double]*,
+                  double *,
+                  int,
+                  int,
+                  double *,
+                  RF_params) except +
+
+    cdef void predict(cumlHandle& handle,
+                      RandomForestMetaData[float, float] *,
+                      float *,
+                      int,
+                      int,
+                      float *,
+                      bool) except +
+
+    cdef void predict(cumlHandle& handle,
+                      RandomForestMetaData[double, double]*,
+                      double *,
+                      int,
+                      int,
+                      double *,
+                      bool) except +
+
+    cdef RF_metrics score(cumlHandle& handle,
```

Comment:
Change this function to be one parameter per line like the prior ones for consistency